mysql-test/suite/innodb/t/max_trx_no_recovery.test

```diff
@@ -0,0 +1,138 @@
+--source include/have_innodb.inc
+--source include/count_sessions.inc
+# The embedded server tests do not support restarting.
+--source include/not_embedded.inc
+--source include/have_debug.inc
+--source include/have_debug_sync.inc
+
+--let transactions_count=10
+--let perl_inc_file=$MYSQL_TMP_DIR/max_trx_no_recovery.inc
+
+--let $i=$transactions_count
+# We need several tables to avoid blocking on records locking and pages latching
+while($i) {
+--dec $i
+if ($i == 0) {
+--eval CREATE TABLE t_1_$i(a INT primary key) Engine=InnoDB STATS_PERSISTENT=0
+--eval INSERT INTO t_1_$i SET a = 10
+}
+--eval CREATE TABLE t_2_$i(a INT primary key) Engine=InnoDB STATS_PERSISTENT=0
+--eval INSERT INTO t_2_$i SET a = 10
+}
```

Comment:
This might possibly be executed on the server side with an inline procedure that generates SQL. You will find `EXECUTE IMMEDIATE` in `mysql-test/suite/innodb/t/instant_alter_bugs.test`, for example.

I’m not suggesting or requesting that actual change here, because this test is forced to use client-side generated SQL for other reasons.